File: src/components/auth/LoginForm.js
import React from "react";
import useLoginViewModel from "../../viewmodels/auth/LoginViewModel";

const LoginForm = ({ navigate, updateRole }) => {
  const vm = useLoginViewModel();

  const onSubmit = async (e) => {
    e.preventDefault();
    const result = await vm.login();
    if (result) {
      localStorage.setItem("role", result.role);
      updateRole(result.role);
      navigate(result.role === "admin" ? "/admin-dashboard" : "/dashboard");
    }
  };

  return (
    <form
      onSubmit={onSubmit}
      className="bg-white p-6 rounded-lg shadow-md max-w-md mx-auto"
    >
      <h2 className="text-2xl font-semibold mb-4 text-center text-blue-600">
        Connexion
      </h2>
      <input
        type="email"
        placeholder="Email"
        value={vm.email}
        onChange={(e) => vm.setEmail(e.target.value)}
        className="w-full mb-4 px-3 py-2 border rounded-lg"
        required
      />
      <input
        type="password"
        placeholder="Mot de passe"
        value={vm.password}
        onChange={(e) => vm.setPassword(e.target.value)}
        className="w-full mb-4 px-3 py-2 border rounded-lg"
        required
      />
      {vm.error && <p className="text-red-500 mb-4 text-center">{vm.error}</p>}
      <button
        type="submit"
        disabled={vm.loading}
        className="w-full bg-blue-600 hover:bg-blue-700 text-white py-2 rounded-lg"
      >
        {vm.loading ? "Connexion..." : "Se connecter"}
      </button>
    </form>
  );
};

export default LoginForm;

File: lib/components/auth/login_form.dart
// lib/components/auth/login_form.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../viewmodels/auth/login_viewmodel.dart';

class LoginForm extends StatefulWidget {
  final Function(String) onRoleUpdate;

  const LoginForm({super.key, required this.onRoleUpdate});

  @override
  State<LoginForm> createState() => _LoginFormState();
}

class _LoginFormState extends State<LoginForm> {
  final _formKey = GlobalKey<FormState>();

  @override
  Widget build(BuildContext context) {
    final vm = Provider.of<LoginViewModel>(context);

    return Form(
      key: _formKey,
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          const SizedBox(height: 24),

          // Email Field
          TextFormField(
            onChanged: vm.setEmail,
            keyboardType: TextInputType.emailAddress,
            decoration: InputDecoration(
              hintText: 'Email',
              filled: true,
              fillColor: Colors.white,
              border: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFFD1D5DB)),
              ),
              enabledBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFFD1D5DB)),
              ),
              focusedBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFF2563EB), width: 2),
              ),
              contentPadding: const EdgeInsets.symmetric(
                horizontal: 12,
                vertical: 12,
              ),
            ),
            validator: (value) {
              if (value == null || value.isEmpty) {
                return 'Email requis';
              }
              if (!value.contains('@')) {
                return 'Email invalide';
              }
              return null;
            },
          ),
          const SizedBox(height: 16),

          // Password Field
          TextFormField(
            onChanged: vm.setPassword,
            obscureText: true,
            decoration: InputDecoration(
              hintText: 'Mot de passe',
              filled: true,
              fillColor: Colors.white,
              border: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFFD1D5DB)),
              ),
              enabledBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFFD1D5DB)),
              ),
              focusedBorder: OutlineInputBorder(
                borderRadius: BorderRadius.circular(8),
                borderSide: const BorderSide(color: Color(0xFF2563EB), width: 2),
              ),
              contentPadding: const EdgeInsets.symmetric(
                horizontal: 12,
                vertical: 12,
              ),
            ),
            validator: (value) {
              if (value == null || value.isEmpty) {
                return 'Mot de passe requis';
              }
              return null;
            },
          ),

          // Error Message
          if (vm.error != null) ...[
            const SizedBox(height: 16),
            Text(
              vm.error!,
              style: const TextStyle(
                color: Colors.red,
                fontSize: 14,
              ),
              textAlign: TextAlign.center,
            ),
          ],

          const SizedBox(height: 24),

          // Login Button
          ElevatedButton(
            onPressed: vm.loading
                ? null
                : () async {
              if (_formKey.currentState!.validate()) {
                final result = await vm.login();
                if (result != null && mounted) {
                  widget.onRoleUpdate(result['role'] ?? 'student');
                  if (result['role'] == 'admin') {
                    Navigator.pushReplacementNamed(context, '/admin');
                  } else {
                    Navigator.pushReplacementNamed(context, '/home');
                  }
                }
              }
            },
            style: ElevatedButton.styleFrom(
              backgroundColor: const Color(0xFF2563EB),
              foregroundColor: Colors.white,
              minimumSize: const Size(double.infinity, 48),
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(8),
              ),
              elevation: 0,
            ),
            child: vm.loading
                ? const SizedBox(
              height: 20,
              width: 20,
              child: CircularProgressIndicator(
                strokeWidth: 2,
                valueColor: AlwaysStoppedAnimation<Color>(Colors.white),
              ),
            )
                : const Text(
              'Se connecter',
              style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w600,
              ),
            ),
          ),
        ],
      ),
    );
  }
}